aioesphomeapi/_frame_helper.py

```diff
@@ -178,18 +179,35 @@ async def _read_frame(self) -> bytes:
         _LOGGER.debug("Received frame %s", frame.hex())
         return frame
 
-    async def perform_handshake(self) -> None:
+    async def perform_handshake(self, expected_name: Optional[str]) -> None:
         await self._write_frame(b"")  # ClientHello
         prologue = b"NoiseAPIInit" + b"\x00\x00"
+
         server_hello = await self._read_frame()  # ServerHello
         if not server_hello:
             raise HandshakeAPIError("ServerHello is empty")
+
+        # First byte of server hello is the protocol the server chose
+        # for this session. Currently only 0x01 (Noise_NNpsk0_25519_ChaChaPoly_SHA256)
+        # exists.
         chosen_proto = server_hello[0]
         if chosen_proto != 0x01:
             raise HandshakeAPIError(
                 f"Unknown protocol selected by client {chosen_proto}"
             )
 
+        # Check name matches expected name (for noise sessions, this is done
+        # during hello phase before a connection is set up)
+        # Server name is encoded as a string followed by a zero byte after the chosen proto byte
+        server_name_i = server_hello.find(b"\0", 1)
```

Comment:
Not sure how guarded (a￼gainst bad data) you need the API to be but you may want to handle the cases of zero length message and string that is \0 only.

Reply:
Both are already handled - `find` returns -1 when the subsequence is not found. So both zero-length messages and `b"\0"` will return -1